tfx/orchestration/pipeline.py

```diff
@@ -189,3 +200,23 @@ def components(self, components: List[base_node.BaseNode]):
     # has all its dependencies visited.
     if len(self._components) < len(deduped_components):
       raise RuntimeError('There is a cycle in the pipeline')
+
+  def set_executor(self, component_id: Text, executor_factory: Type[FakeComponentExecutorFactory]) -> None:
```

Comment:
I think that it would be generally better to pass a Factory instance as an argument, rather than a type. If executor_factory is either FakeComponentExecutorFactory instance or a real executor class type, it would be very helpful. For example, we might be able to use a custom made executor class for one component and use mocks for other components.

It means that we might want the type of `executor_factory` is `FakeComponentExecutorFactory` OR `Type[BaseExecutor]`.
But IIUC, what we really use is executor_spec (For example, `FakeExecutorClassSpec`), So I'd like to suggest change argument to ExecutorSpec, then we might be able to assign it directly to the component.